### DISCUSSION TITLE
COLMAP rotation matrix vs drone data

### DISCUSSION THREAD

**Discussion by stinastrandberg** [colmap/colmap#2468] [Q&A]
Hi!

I'm doing a project comparing COLMAP to using a drone's own pose data when creating NeRFs. I'm quite new to this and need some help with the coordinate systems.

My drone has pitch, roll and yaw data, and the coordinate system they use is having x go through the front of the aircraft, y through the side and z down (consistent with aircraft body coordinate systems). My roll, pitch and yaw data is rotation around x, y and z axis respectively. 

When I calculate my rotation matrices using this data, they look very different to that of COLMAP for the same frame, but I don't understand exactly why. If COLMAP uses opencv as a camera, how do I make sure my rotation matrices are relative to the same coordinate system? How does COLMAP decide on an origin and how can I find it? 

To summarize, how can I use my drone data instead and get the correct rotation matrix that I can use instead of COLMAP as input?

Let me know if any additional information is needed in order to answer this question. Thank you in advance! :)

**Comment by tmnku** [comment]
So what you're saying is that you get poses from your drone and poses from a reconstruction (using images captured by the drone) and they look different?
Possible reasons for that could be:
- Poses are in different coordinate systems. As long as you don't align your reconstruction to world coordinates its origin and orientation is random. Also if you are using a single camera they have different scales.
- Poses in colmap (the ones in image.bin and images.txt) transform from world to camera. What you get out of your drone is likely the other way around.
- When dealing with roll, pitch and yaw you get different rotations depending on the convention of rotation order, so make sure you are using the correct one
